ansible/playbooks/tasks/azure-cluster-bootstrap.yaml

```diff
@@ -9,10 +9,46 @@
   loop:
     - socat
     - resource-agents
-#    
-#- name: Get package information
-#  package_facts:
-#    manager: "auto"
+
+- name: Get the status of all extensions
+  ansible.builtin.command: SUSEConnect -s
+  register: sc_status
+  changed_when: no
+
+- name: Set public cloud extension fact
+  ansible.builtin.set_fact:
+    public_cloud_module: "{{ (sc_status.stdout | from_json | json_query(jmesquery))[0] }}"
+  vars:
+    jmesquery: "[? identifier==`sle-module-public-cloud`].status"
+
+- name: debug
+  ansible.builtin.debug:
+    var: public_cloud_module
+    
+
+- name: Activate public cloud module [SLES 12]
+  ansible.builtin.command: SUSEConnect -p sle-module-public-cloud/12/x86_64 # Only works on x86_64 for now!  
+  when:
+    - ansible_distribution_major_version == '12'
+    - public_cloud_module != 'Registered'
+
+- name: Activate public cloud module [SLES 15]
+  ansible.builtin.command: "SUSEConnect -p sle-module-public-cloud/{{ ansible_distribution_version }}/x86_64" # Only works on x86_64 for now!  
+  when:
+    - ansible_distribution_major_version == '15'
+    - public_cloud_module != 'Registered'
+
+# TODO: Version Control
+- name: Ensure Azure Python SDK and Azure Identity python modules are installed
+  community.general.zypper:
+    name: "{{ item.package }}"
+    state: present
+  loop:
+    # For SLES 12 a we need to force a downgrade of python-azure-core see https://www.suse.com/support/kb/doc/?id=000020716
+    - { 'sles' : '12', 'package' : ['python-azure-mgmt-compute', 'python-azure-identity', 'python-azure-core==1.9.0-2.3.4'] } 
+    - { 'sles' : '15', 'package' : ['python3-azure-mgmt-compute', 'python3-azure-identity'] }
+  when: item.sles == ansible_distribution_major_version
+  
```

Comment:
Some additional set-up actions.  Some are for both scenarios, some are just for Azure fencing.